### ISSUE TITLE
Potential memory leak during tomcat shutdown,

### ISSUE THREAD

**Issue by vguna** [googlemaps/google-maps-services-java#232]
Hi.

I'm using 0.17.1 to lookup address geolocations within a webapp.
On shutdown of tomcat, it complains about non terminated threads:

```java
WARNUNG: The web application [acme-backend] appears to have started a thread named [RateLimitExecutorDelayThread] but has failed to stop it. This is very likely to create a memory leak. Stack trace of thread:
 sun.misc.Unsafe.park(Native Method)
 java.util.concurrent.locks.LockSupport.park(LockSupport.java:175) java.util.concurrent.locks.AbstractQueuedSynchronizer$ConditionObject.await(AbstractQueuedSynchronizer.java:2039)
 java.util.concurrent.LinkedBlockingQueue.take(LinkedBlockingQueue.java:442)
 com.google.maps.internal.RateLimitExecutorService.run(RateLimitExecutorService.java:89)
 java.lang.Thread.run(Thread.java:745)
```

Any idea how to address this?

**Comment by domesticmouse** [comment]
The finalizer for the context needs to shut down the `RateLimitExecutorDelayThread` thread. Patches welcomed =)

**Comment by domesticmouse** [comment]
This should be fixed by #272
